src/panels/ConfigPanel/ConfirmationDialog/ConfirmationDialog.tsx

```diff
@@ -0,0 +1,203 @@
+import { ConfirmationModal } from "@canonical/react-components";
+import { useParams } from "react-router-dom";
+import usePortal from "react-useportal";
+
+import type { EntityDetailsRoute } from "components/Routes";
+import SecretLabel from "components/secrets/SecretLabel";
+import useAnalytics from "hooks/useAnalytics";
+import useCanManageSecrets from "hooks/useCanManageSecrets";
+import { type SetError } from "hooks/useInlineErrors";
+import { useGrantSecret, useSetApplicationConfig } from "juju/api-hooks";
+import type { usePanelQueryParams } from "panels/hooks";
+import { ConfirmType as DefaultConfirmType } from "panels/types";
+import { getModelSecrets } from "store/juju/selectors";
+import { useAppSelector } from "store/store";
+
+import ChangedKeyValues from "../ChangedKeyValues";
+import type { Config, ConfigQueryParams, ConfirmTypes } from "../types";
+import { ConfigConfirmType } from "../types";
+import { getRequiredGrants } from "../utils";
+
+import { InlineErrors, Label } from "./types";
+
+type Props = {
+  confirmType: ConfirmTypes;
+  queryParams: ConfigQueryParams;
+  setEnableSave: React.Dispatch<React.SetStateAction<boolean>>;
+  setSavingConfig: React.Dispatch<React.SetStateAction<boolean>>;
+  setConfirmType: React.Dispatch<React.SetStateAction<ConfirmTypes>>;
+  setInlineError: SetError;
+  config: Config;
+  handleRemovePanelQueryParams: ReturnType<typeof usePanelQueryParams>[2];
+};
+
+const ConfirmationDialog = ({
+  confirmType,
+  queryParams,
+  setEnableSave,
+  setSavingConfig,
+  setConfirmType,
+  setInlineError,
+  config,
+  handleRemovePanelQueryParams,
+}: Props): JSX.Element | null => {
+  const { Portal } = usePortal();
+  const { userName, modelName } = useParams<EntityDetailsRoute>();
+  const { entity: appName, modelUUID } = queryParams;
+  const grantSecret = useGrantSecret(userName, modelName);
+  const setApplicationConfig = useSetApplicationConfig(userName, modelName);
+  const canManageSecrets = useCanManageSecrets();
+  const sendAnalytics = useAnalytics();
+  const secrets = useAppSelector((state) => getModelSecrets(state, modelUUID));
+
+  async function _submitToJuju() {
+    if (!modelUUID || !appName) {
+      return;
+    }
+    setSavingConfig(true);
+    const response = await setApplicationConfig(appName, config);
+    const errors = response?.results?.reduce<string[]>((collection, result) => {
+      if (result.error) {
+        collection.push(result.error.message);
+      }
+      return collection;
+    }, []);
+    setSavingConfig(false);
+    setEnableSave(false);
+    setConfirmType(null);
```

Comment:
This call might be redundant, as we already set the config type to `null` on line 106.